repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilterChain.scala

```diff
@@ -84,7 +85,7 @@ class ReposeFilterChain(val filterChain: List[FilterContext], originalChain: Fil
     var conditionallyWrappedRequest = request
     var conditionallyWrappedResponse = response
 
-    val doLogging = IntrafilterLog.isTraceEnabled
+    val doLogging = IntrafilterLog.isTraceEnabled && Option(MDC.get(TRACE_REQUEST)).isDefined
```

Comment:
I'm not really a fan of checking if intrafilter is enabled or not. It seems pointless to do. If they sent the header they've expressed that they want the info.

Reply:
I look at it from the other side. If we don't check if the logger is set to trace, then we set up the request stream for buffering (which includes copying the entire request body) whenever the header is present. If the header is present and the log level in configured below trace, though, we will never see the information that we set ourselves up to see. So it seems pointless to me to do work that essentially gets dropped on the ground.

That being said, we also add a UUID header as part of the intrafilter setup, so there's some argument to be made about the using sending the header just to get the UUID added... but that seems a little silly.

I can drop that part of the check if you would like (both you and Bill pointed it out), but I do see a reason not to drop it.

Reply:
Eh, you've told us you want to debug this request, i don't care if we pointlessly add overhead. It just adds complication, especially when in this case maybe they wanted timing numbers, but they won't get them because you made that part of the condition to get into the metrics collection.

Reply:
Actually that brings an interesting point, we've added overhead to the filter timing with the pointless intrafilter work. That suggests that code wise we need to be a little more discerning about when we do different things.

Reply:
To that point, I would argue that you've also told us you don't want to debug any request by turning the intrafilter logger level down below trace. What's more, if you expect intrafilter logging information simply because you sent the header, you might be surprised when you don't see it.

In fact, dropping the logger level check also adds complication. That takes us from two possible states to three. The current states are: do intrafilter logging with support code or do not do intrafilter logging with support code. The states after this change would be: do not do intrafilter logging with support code, do support code but not intrafilter logging, and do intrafilter logging with support code.

As for timing numbers, I'm not sure I follow. None of the code gated by this check affects the filter timing logging?

Reply:
I don't think we have added overhead to the filter timing with the intrafilter work? The timing code just wraps the "requestProcess" code, which is where we call `doFilter`. Then the intrafilter code wraps all of that. Since the intrafilter code is on the outside, though, the timing should not include it, and should still be correct?

Here's an example of function composition to show that:
```
scala> val f: (String) => String = s => s + "f"
f: String => String = <function1>

scala> val g: (String) => String = s => s + "g"
g: String => String = <function1>

scala> f.compose(g).apply("hello")
res2: String = hellogf
```

Reply:
except the numbers include everything down stream which will have the intrafilter overhead. Also it looked like whether or not to log the timing was blocked in an if that included the intrafilter logging being on as a check. that could just be github misdirecting me

Reply:
yeah, it was github lying to me. so I think i've come around to leaving it as is.

Reply:
Yeah, I'm still not super satisfied that we had to resort to code to get the logging we want, but I wasn't able to come up with any better solutions...